src/MOI_wrapper.jl

```diff
@@ -218,6 +218,63 @@ function MOI.optimize!(model::Optimizer)
 end
 
 
+const QP_SET_TYPES = (
+    MOI.GreaterThan{Float64},
+    MOI.LessThan{Float64},
+    MOI.EqualTo{Float64},
+    MOI.Interval{Float64}
+)
+
+const QP_FUNCTION_TYPES = (
+    MOI.SingleVariable,
+    MOI.ScalarAffineFunction{Float64}
+)
+
+const QP_OBJECTIVE_TYPES = (
+    MathOptInterface.ScalarAffineFunction{Float64},
+    MathOptInterface.ScalarQuadraticFunction{Float64},
+)
+
+"""
+    backward(model::Optimizer, params...)
+
+Wrapper method for the backward pass. Checks the model/problem type and dispatches a call to
+[`backward_quad`](@ref) or [`backward_conic`](@ref) accordingly.
+"""
+function backward(model::Optimizer, params...)
+    # check if the model is a QP
+    isQP = true
+
+    # check constraints
+    con_types = MOI.get(model.optimizer, MOI.ListOfConstraints())
+    for types in con_types
+        func = types[1]
+        set = types[2]
+
+        if !in(func, QP_FUNCTION_TYPES) || !in(set, QP_SET_TYPES)
+            isQP = false
+        end
+    end
+
+    # check objective
```

Comment:
why is the objective check commented out?

Reply:
as @blegat suggested, i thought its good to determine backend type using constraints only

Reply:
anyway switched back to prev check, was failing on some tests :D